diff --git a/node_modules/thermo_beacon_ws08/thermo_beacon_ws08.js b/node_modules/thermo_beacon_ws08/thermo_beacon_ws08.js
index 5dd9e01..efc1051 100644
--- a/node_modules/thermo_beacon_ws08/thermo_beacon_ws08.js
+++ b/node_modules/thermo_beacon_ws08/thermo_beacon_ws08.js
@@ -12,7 +12,12 @@ class ThermoBeaconWS08 {
       ble_ctl.send (ble_mac);
 
       ble_ctl.on ('message', function (json) {
-        let buf = Buffer.from (json.message.ManufacturerData['17']);
+        let buf;
+        try {
+          buf = Buffer.from (json.message.ManufacturerData['17']);
+        } catch (error) {
+          callback (error, null)
+        } 
         if (buf.byteLength == 18) {
           ble_ctl.kill ('SIGINT');
           ble_ctl.send (ble_mac);
